### ISSUE TITLE
restrict day-of-month value so that only valid dates can be represented

### ISSUE THREAD

**Issue by wbolster** [wbolster/temporenc#21]
dates like YYYY-02-31 should be rejected since february never has 31 days.

the spec should be explicit about this and declare such values invalid.

**Comment by marshallpierce** [comment]
I'm not sure this is really a good responsibility for temporenc to shoulder, since there's no way we could be consistent and also apply rules for leap seconds, for instance. Or, what about the minutes that don't exist when daylight saving time kicks in, unless you're in one of the areas that don't honor it, or one of the areas inside *those* areas that *do* honor it... (Arizona has no DST, but the Navajo nation does, but the Hopi nation inside that doesn't.)

**Comment by lifthrasiir** [comment]
@marshallpierce 

> Or, what about the minutes that don't exist when daylight saving time kicks in, unless you're in one of the areas that don't honor it, or one of the areas inside *those* areas that *do* honor it...

Is it ever a responsibility of Temporenc? I mean, Temporenc in the current form does not encode the geographical and political time "zone" but only the time zone offset, so it seems orthogonal to this issue. The date being not (proleptic) Gregorian sounds much worse.

**Comment by wbolster** [comment]
yeah, imo basic sanity checks to assert that values actually match the (proleptic) gregorian calendar model should be part of a spec that encodes that calendar. checks for day-of-month are cheap since those only require some simple lookups and a [simple calculations for leap years](https://tools.ietf.org/html/rfc3339#appendix-C).
